kv/memberlist/memberlist_client_test.go

```diff
@@ -894,10 +895,12 @@ func TestJoinMembersWithRetryBackoff(t *testing.T) {
 					if err != nil {
 						t.Errorf("failed to start KV: %v", err)
 					}
+					kvsStarted.Done()
```

Comment:
We should probably use `defer kvsStarted.Done()` in this function, to unblock the main test goroutine even if starting of KVs fails.

Reply:
we can to run `kvsStarted.Done()` before running `runClient`. And `t.Errorf` continues the execution of the test

Don't we want to abort the test if a kv cannot start? I changed it so that we abort if the kv store cannot start _and_ call `kvsStarted.Done()` regardless if the kv store started successfully or failed. See https://github.com/grafana/dskit/pull/412/commits/da2d51311ff6bb1e31bce42a08815e569a92162c

Reply:
Thanks, I think that should do it.